src/test/java/KeyProviderTest.java

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import com.google.sps.utility.KeyProvider;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Test Key Provider functions */
+@RunWith(JUnit4.class)
+public class KeyProviderTest {
+
+  private static final String SAMPLE_KEY = "sampleKey";
+  private static final String CAPITALISED_SAMPLE_KEY = "SAMPLEKEY";
+  private static final String INVALID_KEY = "invalidKey";
+  private static final String SAMPLE_VALUE = "sampleValue";
+  private static final String KEYS_JSON = "{\"sampleKey\" : \"sampleValue\"}";
+  private static final File file = new File("src/main/resources/TEST_KEYS.json");
+
+  @Before
+  public void setUp() {
+    // Creates a new file in src/main/resources and writes json content to the file.
+    try {
+      file.getParentFile().mkdirs();
+      file.createNewFile();
+      FileWriter writer = new FileWriter("src/main/resources/TEST_KEYS.json");
+      writer.write(KEYS_JSON);
+      writer.close();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  @After
+  public void tearDown() {
+    // Removes the file created in src/main/resources from calling init()
+    file.delete();
+  }
+
+  @Test
+  public void getSampleKeyValue() throws IOException {
+    // Gets the value of sampleKey which is in src/main/resources/TEST_KEYS.json.
+    // invalidKey is expected.
```

Comment:
Woops I meant `sampleValue`, however, I have deleted the `// xyz is expected.` comments because I don't think they add any value. Thanks for pointing this out!